src/pyhf/pdf.py

```diff
@@ -731,9 +731,10 @@ def logpdf(self, pars, data):
             ):  # force to be not scalar, should we changed with #522
                 return tensorlib.reshape(result, (1,))
             return result
-        except:  # noqa: E722
+        except Exception as excep:
             log.error(
-                f'eval failed for data {tensorlib.tolist(data)} pars: {tensorlib.tolist(pars)}'
+                str(excep)
```

Comment:
quick tip - `repr(excep)` will show the error as well as the message
```
In [1]: try:
   ...:     1/0
   ...: except Exception as e:
   ...:     print(repr(e))
   ...: 
ZeroDivisionError('division by zero')

In [2]: try:
   ...:     1/0
   ...: except Exception as e:
   ...:     print(str(e))
   ...: 
division by zero
```

Reply:
Nice. Thanks for the good tip, @MarcoGorelli!